generate-secret.py

```diff
@@ -1,41 +1,46 @@
-#!/usr/bin/env python3.6
+#!/usr/bin/env python3
 # -*- coding: UTF-8 -*-
 
 """
-Usage:
+Generates a secret key called <app-name>.key.
 
-    generate-secret.py <app-name>
+Usage:
 
-Generates a secret key called <app-name>.key.
+    ./generate-secret.py <app-name>
 """
 
-import sys
+import argparse
 import secrets
-from pathlib import Path
 import shutil
-import logging
-
-logger = logging.getLogger(__name__)
-
-app_name = Path("redeploy", sys.argv[1])
-# assert app_name.exists()
-
+from pathlib import Path
 
-key_file = Path(sys.argv[1]+ ".key")
 
+DEFAULT_USER = "deploy"
+
+here = Path(__file__).parent
+
+parser = argparse.ArgumentParser(description=__doc__.splitlines()[1])
+parser.add_argument(
+    "app_name",
+    metavar="app-name",
+    help="name of the app you want to deploy; "
+    "must match a key in DEPLOYMENTS in app/configuration.py",
+)
+parser.add_argument(
+    "-u",
+    "--user",
+    default=DEFAULT_USER,
+    help=f"owner of the key [default: {DEFAULT_USER}]",
+)
+parser.add_argument(
+    "-g",
+    "--group",
+    default=DEFAULT_USER,
+    help=f"group of the key [default: {DEFAULT_USER}]",
+)
+args = parser.parse_args()
+
+key_file = here / f"{args.app_name}.key"
```

Comment:
I recommend doing

    os.umask(0o377) # prevent creating a world-readable file

before creating the file; there’s a race condition if doing the `chmod` afterwards.

Reply:
I WAS WONDERING HOW TO DO THIS. I was like... why can't I do this with `open()`? It'd be nice for all of these steps to be atomic, haha!

Reply:
Oops, I guess with my suggested umask, you wouldn’t be able to write to the file to start with? Or does being root override that?